src/Compiler/ES/Node.idr

```diff
@@ -50,26 +50,23 @@ compileExpr :
   (outputDir : String) ->
   ClosedTerm ->
   (outfile : String) ->
-  Core (Maybe String)
+  Core String
 compileExpr c s tmpDir outputDir tm outfile =
   do es <- compileToNode c s tm
      let out = outputDir </> outfile
-     Core.writeFile out es
-     coreLift_ $ chmodRaw out 0o755
-     pure (Just out)
+     writeFile out es
+     handleFileError out $ chmodRaw out 0o755
+     pure out
 
 ||| Node implementation of the `executeExpr` interface.
 executeExpr :
   Ref Ctxt Defs ->
   Ref Syn SyntaxInfo ->
-  (tmpDir : String) -> ClosedTerm -> Core ()
+  (tmpDir : String) -> ClosedTerm -> Core ExitCode
 executeExpr c s tmpDir tm =
-  do let outn = tmpDir </> "_tmp_node.js"
-     js <- compileToNode c s tm
-     Core.writeFile outn js
+  do out <- compileExpr c s tmpDir tmpDir tm "_tmp_node.js"
      node <- coreLift findNode
-     quoted_node <- pure $ "\"" ++ node ++ "\"" -- Windows often have a space in the path.
-     coreLift_ $ system (quoted_node ++ " " ++ outn)
+     system $ "\"" ++ node ++ "\" " ++ out -- Windows often have a space in the path.
```

Comment:
Why does `escapeArg` not enclose in quotes in Windows?
https://github.com/idris-lang/Idris2/blob/ec74792a49bf4d22509172d1f03d153ffca1b95c/libs/base/System/Escape.idr#L10-L14

Reply:
Oh, it escapes spaces, okay
https://github.com/idris-lang/Idris2/blob/ec74792a49bf4d22509172d1f03d153ffca1b95c/libs/base/System/Escape.idr#L21

Reply:
Maybe, it worth to add a comment nearby about this to make it clear for the future readers

Reply:
CI on windows fails if I replace quoting with `escapeArg`. It's looks like problem with `escapeArg`

Reply:
> CI on windows fails if I replace quoting with `escapeArg`. It's looks like problem with `escapeArg`

This looks like a Node.js problem https://github.com/nodejs/node/issues/56146

Reply:
I have tried to understand the reasons for this behavior and detailed it in #3436